### DISCUSSION TITLE
Polling until data is ready

### DISCUSSION THREAD

**Discussion by andresespinosapc** [TanStack/query#3755] [Q&A]
I have an endpoint that I have to request multiple times until the data is ready. When the data is not ready yet, it returns no data and a polling token to keep requesting the endpoint until the data is ready. When the data is ready, if I want to get the data again I have to start the polling again.

The behavior I'd want is that the data is only saved in the cache after the polling is complete. I can't use refetch to make the polling, because if I want to get the data again it will become `undefined` until the polling is completed again.

How would you handle this case?

**Comment by TkDodo** [comment]
refetchInterval can be a function so that you can do exactly this: polling until some condition is met.
